codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,219 @@
+            "NOSCRIPT No matching script. Please use EVAL.",
+            "OOM command not allowed when used memory > 'maxmemory'.",
+            "READONLY You can't write against a read only slave.",
+            "WRONGTYPE Operation against a key holding the wrong kind of value",
+    };
+
+    private static final LongObjectHashMap<IntegerRedisMessage> INTEGER_CACHE =
+            new LongObjectHashMap<IntegerRedisMessage>();
+
+    private static PreloadStringsCache preloadSimpleString;
+    private static PreloadStringsCache preloadErrors;
+
+    static {
+        setPreloadStrings(RedisMessageType.SIMPLE_STRING, DEFAULT_PRELOAD_SIMPLE_STRINGS);
+        setPreloadStrings(RedisMessageType.ERROR, DEFAULT_PRELOAD_ERRORS);
+        for (int i = 0; i < 128; i++) {
+            INTEGER_CACHE.put(i, new IntegerRedisMessage(i));
+        }
+    }
+
+    private RedisMessageFactory() {
+    }
+
+    /**
+     * Set preload strings. Users can override preload strings before using {@link RedisMessageFactory}.
+     *
+     * @param type one of {@link RedisMessageType#SIMPLE_STRING} or {@link RedisMessageType#ERROR}.
+     * @param messages an array of String to preload.
+     */
+    public static void setPreloadStrings(RedisMessageType type, String... messages) {
+        ObjectUtil.checkNotNull(messages, "messages");
+        if (type == RedisMessageType.SIMPLE_STRING) {
+            preloadSimpleString = convertPreloadsCache(RedisMessageType.SIMPLE_STRING, messages);
+        } else if (type == RedisMessageType.ERROR) {
+            preloadErrors = convertPreloadsCache(RedisMessageType.ERROR, messages);
+        } else {
+            throw new RedisCodecException("not support type: " + type);
+        }
+    }
+
+    private static PreloadStringsCache convertPreloadsCache(RedisMessageType type, String... messages) {
+        Map<String, StringRedisMessage> stringsCache = new HashMap<String, StringRedisMessage>();
+        Map<ByteBuf, StringRedisMessage> byteBufCache = new HashMap<ByteBuf, StringRedisMessage>();
+        for (String message : messages) {
+            ByteBuf byteBuf = Unpooled.unmodifiableBuffer(
+                    Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8))));
+            StringRedisMessage preloadMessage = new StringRedisMessage(type, message);
+            stringsCache.put(message, preloadMessage);
+            byteBufCache.put(byteBuf, preloadMessage);
+        }
+        return new PreloadStringsCache(stringsCache, byteBufCache);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static StringRedisMessage createSimpleString(String content) {
+        ObjectUtil.checkNotNull(content, "content");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "content" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1192)